
package fr.proto.logger;

option cc_generic_services = true;

enum log_level {
    zero    = 0;
    error   = 1;
    warning = 2;
    info    = 3;
    debug   = 4;
}

message empty { }

message log_req {
    optional log_level level = 1;
    optional string    text  = 2;
}

message set_level_req {
    optional log_level level = 1;
}

message get_level_res {
    optional log_level level = 1;
}

message subscribe_res {
    optional uint32 async_op_id  = 1;
}

message write_data {
    optional log_level level    = 1;
    optional fixed64   microsec = 2;
    optional string    text     = 3;
}

service instance {
    rpc send_log( log_req )         returns ( empty );
    rpc set_level( set_level_req )  returns ( empty );
    rpc get_level( empty )          returns ( get_level_res );

    rpc subscribe( empty )          returns ( subscribe_res );
    rpc unsubscribe( empty )        returns ( empty );
}

